src/snowflake/snowpark/stored_procedure.py

```diff
@@ -464,6 +464,7 @@ def register(
         *,
         statement_params: Optional[Dict[str, str]] = None,
         source_code_display: bool = True,
+        native_app_params: Optional[Dict[str, Any]] = None,
```

Comment:
This will become a public API but a user can't use it for now. So it's better we hide it. You can consider using the `**kwargs` and remove the docstring of native_app_params